File: JavaScript/[1]Programmers/코딩테스트 고득점 Kit/완주하지못한선수.js
const solution = (participant, completion) => {

    let map = new Map();
    let answer = "";

    for(let people of participant) {
        if(map.get(people)){
            map.set(people,map.get(people)+1);
        } else {
            map.set(people,1);
        }
    }

    for(let person of completion) {
        if(map.get(person)) {
            map.set(person, map.get(person)-1);
        }
    }

    for(let people of participant){
        if(map.get(people) === 1) {
            answer = people;
        }
    }

    return answer;
}

// const participant =® ["l®eo", "kiki", "eden"];
// const completion = ["eden", "kiki"];

const participant = ["mislav", "stanko", "mislav", "ana"];
const completion = ["stanko", "ana", "mislav"];

console.log(solution(participant, completion));

File: python/[1]Programmers/코딩테스트 고득점 Kit/완주하지못한선수.py
def solution(participant, completion):  
  participant.sort()
  completion.sort()
  for i in range(len(completion)) :
    if participant[i] != completion[i] :
      return participant[i]
  
  return participant[len(participant)-1]



participant_1 = ["leo", "kiki", "eden"]
completion_1 = ["eden", "kiki"]
participant_2 = ["marina", "josipa", "nikola", "vinko", "filipa"]
completion_2 = ["josipa", "filipa", "marina", "nikola"]
participant_3 = ["mislav", "stanko", "mislav", "ana"]
completion_3 = ["stanko", "ana", "mislav"]


solution(participant_1, completion_1) # leo
solution(participant_2, completion_2) # vinko
solution(participant_2, completion_2) # mislav